src/babel.js

```diff
@@ -27,39 +29,63 @@ module.exports = function plugin(args) {
   }
   const { types: t, template } = args
 
+  // Prevent our own rollup compilation substituting this
+  const processEnvNodeEnv = 'process.' + 'env.' + 'NODE_ENV'; // eslint-disable-line no-useless-concat
+
   const buildRegistration = template(
     'reactHotLoader.register(ID, NAME, FILENAME);',
     templateOptions,
   )
   const headerTemplate = template(
-    `(function () {
-       var enterModule = require('react-hot-loader').enterModule;
-       enterModule && enterModule(module);
-     }())`,
+    `
+(function () {
+  // Allow tree shaking to drop the react-hot-loader module by wrapping with a conditional that DefinePlugin can replace
```

Comment:
lets move this comment above template, or you will add it to the every file processed.